src/insights_client/tests/test_motd.py

```diff
@@ -0,0 +1,101 @@
+import os
+import pathlib
+import tempfile
+import typing
+import unittest
+import unittest.mock
+
+import pytest
+
+import insights_client
+
+
+class MockFS:
+    """Mock filesystem.
+
+    Ensures that the files required for the MOTD manipulation are created
+    in a temporary directory instead of working with actual system files.
+    """
+
+    _dir: tempfile.TemporaryDirectory
+
+    def __init__(self):
+        self._dir = tempfile.TemporaryDirectory(prefix="test-chroot-")
+
+        paths: typing.Dict[str, str] = {
+            "insights_client.MOTD_SRC": "/etc/insights-client/insights-client.motd",
+            "insights_client.MOTD_FILE": "/etc/motd.d/insights-client",
+            "insights_client.REGISTERED_FILE": "/etc/insights-client/.registered",
+            "insights_client.UNREGISTERED_FILE": "/etc/insights-client/.unregistered",
+        }
+
+        self.patches: typing.List[unittest.mock.patch] = []
+        for target, path in paths.items():
+            mocked_path = f"{self.chroot}{path}"
+            pathlib.Path(mocked_path).parent.mkdir(parents=True, exist_ok=True)
+            patch = unittest.mock.patch(target, mocked_path)
+            self.patches.append(patch)
+            patch.start()
+
+        with (self.chroot / "etc/insights-client/insights-client.motd").open("w") as f:
+            f.write("This is the content of the MOTD file.\n")
+
+    @property
+    def chroot(self) -> pathlib.Path:
+        return pathlib.Path(self._dir.name)
+
```

Comment:
I thought up a way how to get the paths from the patches in a more cocsise may. I am not entirely sure about the semantics, but it doesn’t feel completely off either.

```suggestion


    def __getitem__(self, key):
        self.patches[key].new
```

A magic method like this would allow calls like `mock_fs["insights_client.MOTD_FILE"]`. Imagine:

```python
assert not mock_fs["insights_client.MOTD_FILE"].exists()
```

Reply:
It's getting complicated. It is not a production code. I prefer readability over smartness in tests, it can save you significant amount of time when it goes wrong.

Reply:
Here, though, the internal name only serves as a string key to grab the path [defined](https://github.com/m-horky/insights-client/blob/d5a2300b8eda44d6607892307a67107ad5dfc14b/src/insights_client/tests/test_motd.py#L25) here in the tests.

Reply:
The whole [MockFS](https://github.com/RedHatInsights/insights-client/blob/4cab056f37c61d195e6f2567fa0363644caf235a/src/insights_client/tests/test_motd.py#L13) class with its seven methods brings already a great amount of smartness, and thus reducing streamlineliness.

Reply:
> Here, though, the internal name only serves as a string key to grab the path [defined](https://github.com/m-horky/insights-client/blob/d5a2300b8eda44d6607892307a67107ad5dfc14b/src/insights_client/tests/test_motd.py#L25) here in the tests.

That's still missing the point of what I wrote above: the fact that there are mocks to redirect the file operation is a semi-internal implementation detail of the test. The point of the test is checking the resulting paths (that they exist, or what they are, etc), and thus it is correct to use their names in checks/assertions. Once the mocks are set up, from the POV of the tests they should "not be seen" (other than using mock_fs for checks).

So no to your proposal: using the names of the mocks in tests is logically wrong.